### ISSUE TITLE
Source code archive checksum different for tagged release

### ISSUE THREAD

**Issue by dd86k** [gitbucket/gitbucket#2186]
## Issue
**Impacted version**: 4.29.0 (release). All plugins up to date, although I think core is affected

**Deployment mode**: Behind nginx without any issues on that regard.

## Problem description

Like Issue #1333, as it should be fixed in JGit 4.7.0 and GitBucket 4.12.0, I'm experiencing a symptom where every time I download `Source code` of any tagged released, the checksum (sha256) differs every second (assuming the timestamp is generated on the fly).

To reproduce the issue, try the following command (replace upper-cased words):
```
curl https://GITBUCKET/USER/PROJECT/archive/TAG.tar.gz | sha256sum
# or if you want to try my live example
curl https://git.dd86k.space/dd86k/ddcpuid/archive/v0.10.0.tar.gz | sha256sum
```

Despite making a manual tar/zip source file is not affected by this, the URL generated (`assets/*`) is not as pretty as `archive/tag.tar.gz`, e.g.:
```
curl https://git.dd86k.space/dd86k/ddcpuid/releases/v0.10.0/assets/1541393548195Yr5pd0WuIq | sha256sum
```

Am I perhaps missing a critical piece of configuration?

**Comment by takezoe** [comment]
@dd86k Could you try zip format?

**Comment by dax-dev** [comment]
just tested it with dd86k's archive, zip format is also giving different sha256sum output for each run
`curl https://git.dd86k.space/dd86k/ddcpuid/archive/v0.10.0.zip | sha256sum`

**Comment by dd86k** [comment]
If it helps, for example, the time field in ZIP files do change (and so I assume the date does as well). I wonder if it's possible to set those as 0 -- if permitted even.

**Comment by takezoe** [comment]
@dd86k @dax-dev Thanks for confirmation. The time field of archive entries is the cause of difference.

I remembered that it was a bug in JGit's archive functionallity originally. I sent a [patch](https://git.eclipse.org/r/#/c/91116/) to JGit and it was merged. After that, we modified GitBucket to not use JGit's archive functionallity to support partial export of repository. We had to implement archiving by ourselves.

I created a pull request #2192 to fix this problem by setting the commit datetime of the exported revision to archive entries.
